src/vm.rs

```diff
@@ -90,10 +92,27 @@ impl<VCpuType: VirtualCPU> UhyveVm<VCpuType> {
 	pub fn new(kernel_path: PathBuf, params: Params) -> HypervisorResult<UhyveVm<VCpuType>> {
 		let memory_size = params.memory_size.get();
 
+		// TODO: Use random address
+		let guest_address = arch::RAM_START;
+
 		#[cfg(target_os = "linux")]
-		let mem = MmapMemory::new(0, memory_size, arch::RAM_START, params.thp, params.ksm);
+		let mem = MmapMemory::new(
+			0,
+			memory_size,
+			GuestPhysAddr::new(guest_address),
```

Comment:
`arch::RAM_START` was previously, like the constants, an object representing address. This was deliberately changed, as I plan to integrate the function [generate_start_address](https://github.com/hermit-os/uhyve/pull/711/files#diff-3bc51552cab41d1a2dbf07842cb438088563f6134a9c69a266dfd0d79b631495R183) which returns a `u64`. I am not sure if it would be best to make it return a `GuestPhysAddr` instead instead of changing the type of `arch::RAM_START`.

Given my consistent approach of converting some `GuestVirtAddr`'s into `u64` - as we do not expect to use them as addresses anyways, but only as a point of reference / "offset" - so we might as well spare the `.to_u64()` conversion, I decided to do the same thing here.

On the other hand, perhaps the function mentioned above should not be allowed to return integers that can result in valid addresses (or, well, invalid adresses), and by returning an object representing an address, we may have better correctness.

Reply:
Later, we will have to move this entire functionality into `load_kernel`, as we cannot be sure *whether* we can use a guest_address that is not equal to `RAM_START` in the example of `hello_c` during the initalization of `UhyveVm`.